src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -210,5 +214,15 @@ internal abstract class BrokerClient : IBrokerClient {
         public virtual string HandleUrl(string url, CancellationToken ct) {
             return url;
         }
+
+        private async Task CheckMachineOnlineAsync() {
+            if (!Uri.IsFile) {
+                var ping = new Ping();
+                var reply = await ping.SendPingAsync(Uri.Host, 5000);
```

Comment:
ICMP Echo Request may be disabled for security reasons. I don't think we can rely on it outside the corporate network. Even if ping fails, we need to check if we can connect to the socket.